File: Leetcode/Leetcode DailyProblem/February/_28_Left Tree Value.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findBottomLeftValue = function (root) {
  let q = [];
  q.push(root);

  let res = root;
  while (q.length !== 0) {
    let temp = [];
    res = q[0];
    for (let i = 0; i < q.length; i++) {
      if (q[i].left !== null) {
        temp.push(q[i].left);
      }
      if (q[i].right !== null) {
        temp.push(q[i].right);
      }
    }
    q = temp;
  }
  return res.val;
};

File: Leetcode/Leetcode DailyProblem/February/_28_Left_Tree_Value.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    int findBottomLeftValue(TreeNode *root)
    {
        queue<TreeNode *> q;
        q.push(root);
        q.push(NULL);
        TreeNode *ans = root;
        while (!q.empty())
        {
            ans = q.front();
            while (q.front() != NULL)
            {
                TreeNode *temp = q.front();
                q.pop();
                if (temp->left != NULL)
                {
                    q.push(temp->left);
                }
                if (temp->right != NULL)
                {
                    q.push(temp->right);
                }
            }
            q.pop();
            if (!q.empty())
            {
                q.push(NULL);
            }
        }
        return ans->val;
    }
};

int main()
{

    return 0;
}